File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class NotificationOut(BaseModel):
    id: uuid.UUID
    ticket_id: uuid.UUID | None
    message: str
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class UnreadCountOut(BaseModel):
    unread_count: int

File: ios/Models/Notification.swift
import Foundation

/// Backend'deki NotificationOut semasiyla birebir eslesir.
/// NOT: Tip adi "AppNotification" - Swift'in kendi "Notification" tipiyle
/// (NotificationCenter'daki) cakismamasi icin bilerek farkli isimlendirildi.
struct AppNotification: Codable, Identifiable {
    let id: UUID
    let ticketId: UUID?
    let message: String
    let isRead: Bool
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case ticketId = "ticket_id"
        case message
        case isRead = "is_read"
        case createdAt = "created_at"
    }
}

struct UnreadCountResponse: Codable {
    let unreadCount: Int

    enum CodingKeys: String, CodingKey {
        case unreadCount = "unread_count"
    }
}
